kernel/comps/block/src/request_queue.rs

```diff
@@ -45,11 +45,25 @@ impl BioRequestSingleQueue {
     ///
     /// This method will wake up the waiter if a new `BioRequest` is enqueued.
     pub fn enqueue(&self, bio: SubmittedBio) -> Result<(), BioEnqueueError> {
+        // In virtio-blk, the ring buffer has an upper limit for descriptors, which is set to 64.
+        // Here, the upper limit for the number of segments is defined to prevent exceeding the
+        // descriptor limit defined in virtio-blk. Each request in virtio includes an additional
+        // 1 request and 1 response descriptor, therefore this limit is set to 62.
+        const MAX_SEGMENT: usize = 62;
+        assert!(bio.segments().len() <= MAX_SEGMENT);
+
         let mut queue = self.queue.lock();
         if let Some(request) = queue.front_mut() {
             if request.can_merge(&bio) {
-                request.merge_bio(bio);
-                return Ok(());
+                let total_segment_cnt = {
+                    let req_segment_cnt: usize =
+                        request.bios().map(|bio| bio.segments().len()).sum();
+                    req_segment_cnt + bio.segments().len()
```

Comment:
Should avoid doing sum repeatedly by storing the number of segments as a field of `BioRequest`.

Reply:
Done